app/lib/actions/katello/repository/create.rb

```diff
@@ -22,8 +24,10 @@ def plan(repository, clone = false)
               if repository.product.redhat?
                 plan_action(Actions::Candlepin::Environment::AddContentToEnvironment, :view_env_cp_id => view_env.cp_id, :content_id => repository.content_id)
               else
-                content_create = plan_action(Katello::Product::ContentCreate, root)
-                plan_action(Actions::Candlepin::Environment::AddContentToEnvironment, :view_env_cp_id => view_env.cp_id, :content_id => content_create.input[:content_id])
+                unless root.content
```

Comment:
Is it okay to skip these actions like I'm doing here if the root already has a `Content` ?  Otherwise, the `Content` fails to get created during the "Finalize" step.

Reply:
i realized that this rake task handles the situation where the repository is missing in pulp, but doesn't handle if the content is missing in candlepin (chiggins actually hit this recently iirc when testing this script ).  Which is handy because you're assuming here that if force gets passed in we MUST recreate the repository.  If we ever have to handle missing content, this might fall apart.  Maybe we should rename the option to 'force_repo_create' or something like that?  And then we could later pass in force_content_create to handle that particular situation (I think he filed a bug)

Reply:
sorry i wrote a paragraph and didn't answer your question.   I think what you have is fine because we're not handling a missing content object in candlepin.

Reply:
Cool, sounds good to me. I'll change up the naming.